code/game/objects/items/weapons/RCD.dm

```diff
@@ -378,7 +396,210 @@ RCD
 	ammoamt = 160
 
 
-/obj/item/weapon/rcd/admin
+/obj/item/weapon/construction/rcd/admin
 	name = "admin RCD"
 	max_matter = INFINITY
 	matter = INFINITY
+
+
+// Ranged RCD
+
+
+/obj/item/weapon/construction/rcd/arcd
+	name = "advanced rapid-construction-device (ARCD)"
+	desc = "A prototype RCD with ranged capability and extended capacity"
+	max_matter = 300
+	matter = 300
+	delay_mod = 0.6
+	ranged = TRUE
+	icon_state = "arcd"
+
+/obj/item/weapon/construction/rcd/arcd/range_check(atom/A, mob/user)
+	if(!(A in view(7, get_turf(user))))
+		to_chat(user, "<span class='warning'>The \'Out of Range\' light on the [src] blinks red.</span>")
+		return FALSE
+
+
+/obj/item/weapon/construction/rcd/arcd/afterattack(atom/A, mob/user)
+	range_check(A,user)
+	if((isturf(A) && A.density && mode==RCD_DECONSTRUCT) || (isturf(A) && !A.density) || (istype(A, /obj/machinery/door/airlock) && mode==RCD_DECONSTRUCT) || istype(A, /obj/structure/grille) || (istype(A, /obj/structure/window) && mode==RCD_DECONSTRUCT) || istype(A, /obj/structure/girder))
+		user.Beam(A,icon_state="rped_upgrade",time=30)
+	..()
+
+
+
+// RAPID LIGHTING DEVICE
+
+
+
+/obj/item/weapon/construction/rld
+	name = "rapid-light-device (RLD)"
+	desc = "A device used to rapidly provide lighting sources to an area."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rld-5"
+	matter = 200
+	max_matter = 200
+	var/mode = 1
```

Comment:
I knew it, this was your plan all along, baiting me into doing a full RCD refactor like 2 weeks after Goof did a partial one, why didn't anyone tell him to use defines ;_;